src/Mentat.UI/Controllers/MentorController.cs

```diff
@@ -10,20 +10,27 @@ public IActionResult Index()
             return View();
         }
 
-        public IActionResult AddAssignment()
+        public IActionResult AddAssignment(Assignment assignment)
         {
             return View();
         }
 
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public ActionResult SubmitForm(Models.Assignment assignment)
+        public ActionResult SubmitForm(Assignment assignment)
+        {
+            // If form is correct then moves on to the next view 
+            // otherwise view is reloaded and user must retry filling the form
+            if (ModelState.IsValid)
+            {
+                // On Form Validation Success Go To This View
+                return RedirectToAction("Success");
```

Comment:
The success page/action  "/Success/" should be a page of summary where the mentor sees a live summary of the assignment during and after it is being graded. Take it as a loose end that will be used by whoever develops the assignment grading system. It can always be refactored later if they want to change the design